tensorflow/python/kernel_tests/matmul_op_test.py

```diff
@@ -63,9 +64,60 @@ def _GetTransposedMatrices(x, x_name, kwargs):
   else:
     return x
 
-
 class MatMulTest(test_lib.TestCase):
-  pass  # Filled in below
+  def test_bfloat16(self):
```

Comment:
The main function already has a matmul testing template, for which it adds test cases for specific sizes and types.
https://github.com/tensorflow/tensorflow/blob/b2ed6eb035c841df8c78eef183616d8cd374c9fb/tensorflow/python/kernel_tests/matmul_op_test.py#L259

We shouldn't replicate the logic. Can we add the bf16 data type to the `dtype_to_test` list instead?
https://github.com/tensorflow/tensorflow/blob/b2ed6eb035c841df8c78eef183616d8cd374c9fb/tensorflow/python/kernel_tests/matmul_op_test.py#L230

We can add `if`s for things specific to bf16 in `_GetMatMulTest` (the template).
https://github.com/tensorflow/tensorflow/blob/b2ed6eb035c841df8c78eef183616d8cd374c9fb/tensorflow/python/kernel_tests/matmul_op_test.py#L71

For example, this checks for fp16
https://github.com/tensorflow/tensorflow/blob/b2ed6eb035c841df8c78eef183616d8cd374c9fb/tensorflow/python/kernel_tests/matmul_op_test.py#L78-L81

Reply:
Thanks , I will look into it

Reply:
I have updated the coded according to your suggestion.